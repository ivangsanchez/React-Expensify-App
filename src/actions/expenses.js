import { v1 as uuid4} from 'uuid';

/////////////////////AD_ EXPENSE /////////////
export const addExpense = ({ descriptions='',note='',amount=0, createdAt=0 } = {}) => ({
    type:'ADD_EXPENSE',
    expense: {
        id: uuid4(),
        descriptions,
        note,
        amount,
        createdAt
    }
})

/////////////////////REMOVE_EXPENSE //////////////////////////////
 export const removeExpense =({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
})

//////////////////EDIT_EXPENSE //////////////////
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});