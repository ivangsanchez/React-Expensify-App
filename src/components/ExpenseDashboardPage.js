import React from 'react';
import ExpenseList from './ExpenseList';
import ExponseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExponseListFilters/>
        <ExpenseList/>
    </div>
);

export default ExpenseDashboardPage;