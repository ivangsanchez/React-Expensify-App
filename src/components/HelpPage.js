import React from 'react';

const HelpPage = () => (
    <div>
        This is from HELP page!
    </div>
);

export default HelpPage;