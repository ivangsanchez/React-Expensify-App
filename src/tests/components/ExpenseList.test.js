import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('render expensesList with expenses', () =>{
    const wrapper =shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('render expensesList with empty message', () =>{
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});
