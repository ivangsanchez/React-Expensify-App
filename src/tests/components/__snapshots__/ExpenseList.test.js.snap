// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render expensesList with empty message 1`] = `
<div>
  <p>
    No expenses Exists
  </p>
</div>
`;

exports[`render expensesList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={1233}
    createdAt={0}
    descriptions="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={76478}
    createdAt={-345600000}
    descriptions="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={1500}
    createdAt={345600000}
    descriptions="Credit card"
    id="3"
    key="3"
    note=""
  />
</div>
`;
